### ISSUE TITLE
simple eval plugin should respect extension pragmas in file

### ISSUE THREAD

**Issue by codygman** [haskell/haskell-language-server#369]
### Steps to reproduce
Tell us how to reproduce this issue.

Create a stack project and update Main.hs to:

```hs
{-# LANGUAGE OverloadedStrings #-}
module Main where

import Data.ByteString.Lazy.Char8

myByteString :: ByteString
myByteString = "my "

-- >>> myByteString <> "stuff"

main :: IO ()
main = pure ()
```

### Expected behaviour
OverloadedStrings should take effect in the inline code evaluation context resulting in:

```hs
{-# LANGUAGE OverloadedStrings #-}
module Main where

import Data.ByteString.Lazy.Char8

myByteString :: ByteString
myByteString = "my "

-- >>> myByteString <> "stuff"
-- "my stuff"

main :: IO ()
main = pure ()
```

### Actual behaviour

An error indicating the inline evaluation context doesn't respect `OverloadedStrings`. I've also verified it doesn't work with others like `DeriveGeneric` and `TypeApplications`.

```hs
{-# LANGUAGE OverloadedStrings #-}
module Main where

import Data.ByteString.Lazy.Char8

myByteString :: ByteString
myByteString = "my "

-- >>> myByteString <> "stuff"
-- Couldn't match expected type ‘ByteString’ with actual type ‘[Char]’

main :: IO ()
main = pure ()
```

**Comment by pepeiborra** [comment]
Thanks for the bug report. I thought this was already fixed by #322
What version are you using?

**Comment by pepeiborra** [comment]
closing as this is already fixed in 0.4.0
